### PR TITLE
Laravel Valet autocomplete

### PR DESCRIPTION
**Pull request by floq-design**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
NEW laravel valet autocomplete

**What is the current behavior? (You can also link to an open issue here)**
N/A

**What is the new behavior (if this is a feature change)?**
New autocomplete

**Additional info:**
Autocomplete spec for laravel (v8) valet (v2.15.x)

### CODE REVIEW

**Comment by QuiiBz** [comment]
Hello, could you change the base branch to `master`? Looks like the current one is your Wordpress branch

**Comment by floq-design** [comment]
> Hello, could you change the base branch to `master`? Looks like the current one is your Wordpress branch

TBH I should've branched master before I worked on the wp-cli... So I branched master for this valet work and of course I get all the master commits.. I deleted the `wp.ts` file from this branch so just the one file.. 
If I merge back into master then there will be both `wp.ts` & `valet.ts` which happy to do but didn't want to confuse more than already have!

Lemme know..

File: dev/valet.ts [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+      description: "Generate a publicly accessible URL for your project",
+      options: [
+        {
+          name: "--auth",
+          description: "enforce basic auth on tunnel endpoint, 'user:password'",
+        },
+        {
+          name: "--authtoken",
+          description: "ngrok.com authtoken identifying a user",
+        },
```
Comment (mattschrage) [inline dev/valet.ts:449]: Does this take the auth token as an argument?

File: dev/valet.ts [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+            suggestions: [
+              { name: "true", insertValue: "'true'" },
+              { name: "false", insertValue: "'false'" },
+              { name: "both", insertValue: "'both'" },
+            ],
+          },
+        },
+        {
+          name: "--config",
+          description: "path to config files; they are merged if multiple",
```
Comment (mattschrage) [inline dev/valet.ts:464]: Seems like this takes a `path` as an argument.

```
args: {
name: "path",
template: "filepath"
}
```

File: dev/valet.ts [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+        },
+        {
+          name: "--host-header",
+          description:
+            "set Host header; if 'rewrite' use local address hostname",
+        },
```
Comment (mattschrage) [inline dev/valet.ts:470]: Seems like `--host-head` takes an argument?

**Review by mattschrage — CHANGES_REQUESTED** [review]
Thanks so much for contributing @floq-design! 🙌

I could be wrong as I'm not familiar with the `valet` CLI, but it seems like several options are missing arguments.

(You don't need to provide generators or suggestions for them, but if the option doesn't have a corresponding `arg` object then the completions we generate at runtime will be incorrect!)

Otherwise, great work! Excited to get this merged in :)

**Comment by QuiiBz** [comment]
Hello @floq-design, I rebased your branch to remove the previous commits from `wp`. Do you want to resolve Matt's review?

File: dev/valet.ts (hunk 2 above)
Comment (floq-design) [reply dev/valet.ts:464]: Good spot.. suggestion added

File: dev/valet.ts (hunk 3 above)
Comment (floq-design) [reply dev/valet.ts:470]: Yes it does.. args with suggestions added

File: dev/valet.ts (hunk 1 above)
Comment (floq-design) [reply dev/valet.ts:449]: Yes it does.. added arg

**Comment by floq-design** [comment]
> I could be wrong as I'm not familiar with the `valet` CLI...

I'm not familiar with the `share` command for Valet.
These options I believe are based on `ngrok` which I'm also unfamiliar with.
https://ngrok.com/docs#tunnel-definitions

I've reworked a lot of the options based upon the ngrok docs and what the `valet share -h` command shows.
Also added your suggestions - thanks.

**Comment by floq-design** [comment]
Sorry made a right hash of pulling latest build into branch and merging.

**Comment by QuiiBz** [comment]
Are the commits from wp intentional?

**Comment by QuiiBz** [comment]
I again removed the +200 commits. Please pay attention when you pull/merge branches.

**Review by QuiiBz — APPROVED** [review]
Laravel users are going to love this one!

**Pull request merged**